gp2gp-translator/src/integrationTest/java/uk/nhs/adaptors/pss/translator/LargeMessagingIT.java

```diff
@@ -0,0 +1,300 @@
+    private static final boolean OVERWRITE_EXPECTED_JSON = false;
+
+    private static final List<String> STATIC_IGNORED_JSON_PATHS = List.of(
+        "id",
+        "entry[0].resource.id",
+        "entry[0].resource.identifier[0].value",
+        "entry[*].resource.id",
+        "entry[*].resource.subject.reference",
+        "entry[*].resource.patient.reference",
+        "entry[*].resource.performer[0].reference",
+        "entry[*].resource.content[0].attachment.title",
+        "entry[*].resource.content[0].attachment.url",
+        "entry[*].resource.description"
+    );
+    private static final String LOSING_ODS_CODE = "D5445";
+    private static final String WINNING_ODS_CODE = "ABC";
+    private static final String NHS_NUMBER_PLACEHOLDER = "{{nhsNumber}}";
+    private static final String CONVERSATION_ID_PLACEHOLDER = "{{conversationId}}";
+
+    @Autowired
+    private PatientMigrationRequestDao patientMigrationRequestDao;
+
+    @Autowired
+    private MigrationStatusLogService migrationStatusLogService;
+
+    @Autowired
+    private PatientAttachmentLogService patientAttachmentLogService;
+
+    @Qualifier("jmsTemplateMhsQueue")
+    @Autowired
+    private JmsTemplate mhsJmsTemplate;
+
+    @Autowired
+    private ObjectMapper objectMapper;
+
+    @Autowired
+    private FhirParser fhirParserService;
+
+    private String patientNhsNumber;
+    private String conversationId;
+
+
+    @BeforeEach
+    public void setUp() {
+        patientNhsNumber = generatePatientNhsNumber();
+        conversationId = generateConversationId();
+        startPatientMigrationJourney();
+    }
+
+    // Test case 1: UK06 with cid attachment
+    @Test
+    public void handleUk06WithCidAttachment() throws JSONException {
+        sendInboundMessageToQueue("/json/LargeMessage/Scenario_1/uk06.json");
+
+        await().until(this::isEhrExtractTranslated);
+
+        verifyBundle("/json/LargeMessage/expectedBundleScenario1.json");
+    }
+
+    // Test case 2: UK06 with compressed cid attachment
+    @Test
+    public void handleUk06WithCompressedCidAttachmement() throws JSONException {
+        sendInboundMessageToQueue("/json/LargeMessage/Scenario_2/uk06.json");
+
+        await().until(this::isEhrExtractTranslated);
+
+        verifyBundle("/json/LargeMessage/expectedBundleScenario2.json");
+    }
+
+    // Test case 3: UK06 with 1 mid attachment
+    @Test
+    public void handleUk06WithOneMidAttachment() throws JSONException {
+        sendInboundMessageToQueue("/json/LargeMessage/Scenario_3/uk06.json");
+
+        await().until(this::hasContinueMessageBeenRecieved);
+
+        sendInboundMessageToQueue("/json/LargeMessage/Scenario_3/copc.json");
+
+        await().until(this::isEhrExtractTranslated);
+
+        verifyBundle("/json/LargeMessage/expectedBundleScenario3.json");
+    }
+
+    // Test case 4: UK06 with fragment mid attachments
+    @Test
+    public void handleUk06WithFragmentedMids() throws JSONException {
```

Comment:
Have you considered creating parametrised tests, as many of them only differ in json files? You can find an example of using `@ParameterizedTest` in the `ResourceReferenceUtilTest` test class.